RFC-0001-row-id.md

```diff
@@ -0,0 +1,80 @@
+# **RFC 1 for Presto**
+
+## Row ID
+
+Proposers
+
+* Brian Dahmen
+* Devang Shah
+* Elliotte Rusty Harold
+* Serge Druzkin
+* Sreeni Viswanadha
+
+## Related Issues
+
+* Row ID support #22078
+* Binary IDs for Hive partitions #22103
+* Binary rowID type #22053
+* Start adding row\_id hidden column to Hive #22008
+
+
+## Summary
+
+A row ID is a binary key for a row that is globally unique across the warehouse.
+It identifies a specific row at a specific version of a partition’s data. A row
+ID will continue to locate the same row in the warehouse until the underlying
+partition is mutated.
+
+## Background
+
+Queries can produce lists of row IDs that can be saved and used as input to
```

Comment:
If they are saved, and the underlying partitions are not sealed, how is it safe to save it?  I.e. there could be extra data which has been ingested or deleted?

Reply:
I agree this is safe to do in your environment, but what I'm wondering is, because many of these changes will happen in Hive, is it safe to do in Hive?

Query A is writing data to partitions, it takes an hour to complete.
Query B is identifying the deleted rows and putting them in a file.  It starts at the half hour mark and takes 5 minutes.
We are left with query B's files that only contain partial rows.  Now, I want to identify the rows that query B missed because of the concurrent update--how do I do that?

Because this doesn't seem safe, I am wondering if this can be encapsulated purely within an internal connector that speaks in terms of your warehouse, rather than being added to Hive.